packages/number-field/src/vaadin-number-field.js

```diff
@@ -441,6 +460,16 @@ export class NumberField extends InputFieldMixin(SlotStylesMixin(ThemableMixin(E
     super._valueChanged(this.value, oldVal);
   }
 
+  /** @private */
+  __isValidByStep(value) {
```

Comment:
Something fishy about the validation. With the following:
`<vaadin-number-field label="Required" required step="0.3"></vaadin-number-field>`
if I focus the field and hit up arrow 3 times so that the value becomes 0.9, the field becomes invalid when blurred.

Reply:
Thanks, I will add tests for fractional steps.

Reply:
@tomivirkki Fixed, and added `validate()` calls to existing tests for problematic values.

Reply:
The following number field still claims to be invalid if I change the value with arrow keys:
`<vaadin-number-field min="1" step="0.3"></vaadin-number-field>`